Source/AwsCommonRuntimeKit/auth/signing/Signer.swift

```diff
@@ -72,6 +72,105 @@ public class Signer {
             }
         }
     }
+
+    /// Signs a body chunk according to the supplied signing configuration
+    /// - Parameters:
+    ///   - chunk: Chunk to sign
+    ///   - previousSignature: The signature of the previous component of the request: either the request itself for the first chunk,
+    ///                        or the previous chunk otherwise.
+    ///    - config: The `SigningConfig` to use when signing.
+    ///   - allocator: (Optional) allocator to override
+    /// - Returns: Signature of the chunk
+    /// - Throws: CommonRunTimeError.crtError
+    public static func signChunk(chunk: Data,
```

Comment:
I expect the signature to be a byte array here ie `Data`

same for return type.

Any reason choosing it as a String?

Reply:
Also, how would it work if we done't have previous signature. I think previous signature must be optional here.

Reply:
Yes, checked that.

With Transcribe Streaming also, request signature works are previous signature. Feel free to ignore https://github.com/awslabs/aws-crt-swift/pull/151#discussion_r1093602179.

Reply:
For signature, I choosed a string because it matched the C type. I had updated it to Data.

Reply:
I disagree with this change. I'm pretty sure the returned signature should be `String`, and the previousSignature should also be `String`. If it were just `Data` then the user would be like ... am I supposed to base64 encode this or what?

Reply:
I believe, consumers don't have to do anything with String, we can treat it as immutable during the whole request time.